### PR TITLE
Bug 1170463 - [Settings] fix remain jsdoc error r=arthur

### PR DESCRIPTION
**Pull request by yuwen41200**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=76ad807d29870666c849ed9d34499787575ba317

File: apps/settings/js/panels/operator_settings/models/operator_manager.js [hunk 1]
```diff
@@ -142,21 +142,22 @@ define(function(require) {
 
     this.debug('search');
     this._searching = true;
-    return this._connWrapper.search().then((networks = []) => {
-      var operators = networks.map((network) => OperatorItem(network));
-      operators.forEach((operator) => {
+    return this._connWrapper.search().then(function(networks = []) {
```
Comment (gasolin) [inline apps/settings/js/panels/operator_settings/models/operator_manager.js]: why not keep arrow functions?

Comment (gasolin) [inline apps/settings/js/panels/operator_settings/models/operator_manager.js]: ooh seems default argument with arrow cannot be recongnized in current jsdoc `networks = []) =>`

please use 
```
then((networks) => {
networks = networks || [];
var operators =....
```

File: apps/settings/js/panels/operator_settings/models/operator_manager.js [hunk 2]
```diff
@@ -142,21 +142,22 @@ define(function(require) {
...
+      var operators = networks.map(function(network) {
+        return OperatorItem(network); });
+      operators.forEach(function(operator) {
         if (operator.network.state === OperatorItem.STATE.CONNECTED) {
           this._connectedOperatorItem = operator;
         }
       });
       this._operators.reset(operators);
-    }).catch((error) => {
+    }).catch(function(error) {
```
Comment (gasolin) [inline apps/settings/js/panels/operator_settings/models/operator_manager.js]: please keep the origin syntax(preffered), or you need add `.bind(this)` at the end of the function.

File: apps/settings/js/panels/operator_settings/models/operator_manager.js (hunk 1 above)
Comment (gasolin) [inline apps/settings/js/panels/operator_settings/models/operator_manager.js]: & please update jsdoc version to 3.3.2 in `gaia/package.json` first to check if the issue has been solved.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e6f52dfbce06fe58efaf735e3cfc706bf7113730

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e14bf751f5e8e37f9f6cc102f37c8fee58df3b22

File: apps/settings/js/modules/addon_manager.js [hunk 3]
```diff
@@ -568,7 +568,7 @@ define(function(require) {
      * @access public
      * @memberOf AddonManager.prototype
      * @param {App} addon
-     * @returns {Promise String}
+     * @returns {(Promise|String)}
```
Comment (crh0716) [inline apps/settings/js/modules/addon_manager.js]: `|` means the type can be either `Promise` or `String`. However the syntax we need here is "a promise resolved with a string". The closest syntax that I can find is `Promise.<String>`. What do you think?

File: apps/settings/js/modules/apn/apn_list.js [hunk 4]
```diff
@@ -177,7 +177,7 @@ define(function(require) {
      *
      * @access private
      * @memberOf ApnList.prototype
-     * @returns {Promise Array<ApnItem>}
+     * @returns {(Promise|Array<ApnItem>)}
```
Comment (crh0716) [inline apps/settings/js/modules/apn/apn_list.js]: Just realized that we may have to change `Array<ApnItem>` to `Array.<ApnItem>`. Following the previous comment the entire thing should be `Promise.<Array.<ApnItem>>`.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c5a789dab0fc020c846f83808a18218438389f4c

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ce4003ec21ec298eb07d0407ff69e1101a94038b

**Pull request merged**